### DISCUSSION TITLE
[UNCONFIRMED] Copilot extension for VSCode doesn't inherit the --proxy-server flag setting

### DISCUSSION THREAD

**Discussion by Binary-Eater** [community/community#6649] [Copilot Conversations] [answered]
Don't know if this is true or not, but it seems the VSCode extension's requests doesn't get routed through the proxy server specified in the flag used in the invocation of VSCode?

If that is the case, presume the below issue/comment is relevant?

https://github.com/Microsoft/vscode/issues/44941#issuecomment-760111944

**Comment by hsenag** [comment, marked as answer]
That's correct - we don't currently support proxy servers.

**Reply by jnvshubham7** [reply]
**how to use this solutoin please give me brief about that?**
```
copilot-proxy.githubusercontent.com
api.github.com (Also used for Self-hosted runners)
```


This error have shown in vs code when we trying to connect copilot on my collage internet which will use proxy system
`[ERROR] [default] [2022-06-16T05:43:59.030Z] GitHub Copilot could not connect to server. Extension activation failed: "getaddrinfo ENOTFOUND api.github.com"`

how to solve this problem ?
please reply me

**Comment by Dhi13man** [comment]
Any plans to support it any time in the future? Seems pretty important in lots of workspaces and regions. :)

**Reply by hsenag** [reply]
It's on our radar, but I can't promise a timeline at present.

**Reply by aalkhatib9** [reply]
Just wanted to follow up on this, any chance you can provide a timeline now when we can expect to work with proxies? This would be greatly appreciated. Thank you.

**Reply by hsenag** [reply]
Now that we've shipped Copilot as generally available, it's pretty high up our priority list, but we also have holiday season etc coming up. Personally I'd be disappointed if we haven't at least improved the situation by the end of July, but this isn't any kind of promise :-)

**Comment by Dhi13man** [comment]
In the meantime, if anyone else stumbles upon the issue and wants a workaround, the two steps below seem to have fixed it for me.
1. setting the http_proxy, https_proxy [powershell environment variables](https://docs.vmware.com/en/VMware-Cloud-Assembly/services/Using-and-Managing/GUID-58652BA4-2254-4B3B-85A9-6662CB01D1F4.html) and [user environment variables](https://stackoverflow.com/questions/48473330/set-http-proxy-environment-variable-with-domain-user-and-pass-in-windows/48534395).
2. Tunneling my network through a [SoftEther VPNGate connection.](https://www.vpngate.net/en/howto_softether.aspx)

**Reply by leonden** [reply]
Well I just received an email, that my GitHub Account is activated for Copilot. Now I have this issue with the proxy because of my company, so I wanted to take a look at your solution - At least to try if it works. Well long story short the links are blocked by my company.

**Comment by Dhi13man** [comment]
You can try to connect to an Openvpn or Softether VPN Gate connection and
then start vscode. Copilot will start up then tunneling through that VPN
network.

**Reply by jnvshubham7** [reply]
But i have not able to connect vpn by college internet
Any other solution ?
